src/main/java/BoardCopier.java

```diff
@@ -9,8 +7,54 @@ public BoardCopier() {}
      * @param board the board to be cloned
      * @return a clone of the board
      */
-    public Board createCopy(Board board) {
-        // TODO
-        return null;
+
+    private Board newBoard = new Board();
+    private Board oldBoard;
+
+    public BoardCopier(Board board) {
+        this.oldBoard = board;
+    }
+
+    public Board createCopy() {
+        Player[] oldPlayers = this.oldBoard.getPlayers();
+        clonePlayer(oldPlayers);
+        clonePieces();
+        //TODO!!!
+        // Don't entirely understand the use of mirror coordinates
+        // If still needed, will need to implement later
+
+        return this.newBoard;
+    }
+
+    private void clonePieces() {
+        for (int i=0; i<Board.HEIGHT; i++) {
+            for (int j=0; j<Board.WIDTH; j++) {
+                Piece absPiece = oldBoard.getPieceAtAbsCoords(i, j);
+                Piece relPiece = oldBoard.getPieceAtRelCoords(i, j);
+                Piece mirrorPiece = oldBoard.getPieceAtMirrorCoords(i, j);
+
+                if (absPiece != null) {
+                    Piece newPiece = absPiece.copy();
+                    newBoard.setPieceAtAbsCoords(i, j, newPiece);
+                }
+                
+                if (relPiece != null) {
+                    Piece newPiece = relPiece.copy();
+                    newBoard.setPieceAtRelCoords(i, j, newPiece);
+                }
+                 
+                if (mirrorPiece != null) {
+                    Piece newPiece = mirrorPiece.copy();
+                    newBoard.setPieceAtMirrorCoords(i, j, newPiece);
+                } 
+            }
+        }
+    }
+
```

Comment:
add method javadoc. this applies to all following methods missing javadoc as well.